packages/salesforcedx-vscode-apex/src/commands/apexActionController.ts

```diff
@@ -83,12 +90,16 @@ export class ApexActionController {
 
           // Step 7: If the user chose to merge, open a diff between the original and new ESR files
           if (fullPath[0] !== fullPath[1]) {
-            await vscode.commands.executeCommand(
-              'vscode.diff',
-              vscode.Uri.file(fullPath[0]),
-              vscode.Uri.file(fullPath[1]),
-              'Manual Diff of ESR Files'
-            );
+            this.openDiffFile(fullPath[0], fullPath[1], 'Manual Diff of ESR XML Files');
+
+            // If sfdx-project.json contains decomposeExternalServiceRegistrationBeta, also open a diff for the YAML OAS docs
+            if (this.isESRDecomposed) {
+              this.openDiffFile(
```

Comment:
@daphne-sfdc be explicit dealing with calls that return promises
```suggestion
              void this.openDiffFile(
```